bitnami/kafka/3.3/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh

```diff
@@ -240,25 +240,69 @@ kafka_validate() {
 
     if is_boolean_yes "$KAFKA_ENABLE_KRAFT"; then
         if [[ -n "$KAFKA_CFG_BROKER_ID" ]]; then
-            warn "KAFKA_CFG_BROKER_ID Must match what is set in KAFKA_CFG_CONTROLLER_QUORUM_VOTERS"
+            old_IFS=$IFS
+            IFS=','
+            read -a voters <<< "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS"
+            IFS=${old_IFS}
+            broker_id_matched=false
+            for voter in "${voters[@]}"; do
+                if [[ "$voter" == *"$KAFKA_CFG_BROKER_ID"* ]]; then
+                    broker_id_matched=true
+                    break
+                fi
+            done
+
+            if [[ "$broker_id_matched" == false ]]; then
+                warn "KAFKA_CFG_BROKER_ID Must match what is set in KAFKA_CFG_CONTROLLER_QUORUM_VOTERS"
+            fi
         else
             print_validation_error "KRaft requires KAFKA_CFG_BROKER_ID to be set for the quorum controller"
         fi
         if [[ -n "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS" ]]; then
-            warn "KAFKA_CFG_CONTROLLER_QUORUM_VOTERS must match brokers set with KAFKA_CFG_BROKER_ID"
+            if [[ "$broker_id_matched" == false ]]; then
+                warn "KAFKA_CFG_CONTROLLER_QUORUM_VOTERS must match brokers set with KAFKA_CFG_BROKER_ID"
+            fi 
```

Comment:
What about reordering it like this ?

```suggestion
if [[ -z "$KAFKA_CFG_BROKER_ID" ]]; then
    print_validation_error "KRaft requires KAFKA_CFG_BROKER_ID to be set for the quorum controller"
fi
if [[ -z "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS" ]]; then
    print_validation_error "KRaft requires KAFKA_CFG_CONTROLLER_QUORUM_VOTERS to be set"
fi

if [[ -n "$KAFKA_CFG_BROKER_ID" ]] && [[ -n "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS" ]]; then
    old_IFS=$IFS
    IFS=','
    read -r -a voters <<< "$KAFKA_CFG_CONTROLLER_QUORUM_VOTERS"
    IFS=${old_IFS}
    broker_id_matched=false
    for voter in "${voters[@]}"; do
        if [[ "$voter" == *"$KAFKA_CFG_BROKER_ID"* ]]; then
            broker_id_matched=true
            break
        fi
    done

    if [[ "$broker_id_matched" == false ]]; then
        warn "KAFKA_CFG_BROKER_ID must match what is set in KAFKA_CFG_CONTROLLER_QUORUM_VOTERS"
    fi
fi
```